### ISSUE TITLE
Fonts are missing

### ISSUE THREAD

**Issue by jiriks74** [ONLYOFFICE/DocumentServer#1250]
I'm running server on arm64 debian. Fonts work fine, but after intalation of `ttf-mscorefonts-installer` and successful font generation on the onlyoffice side, the fonts are missing on the server. 

What should I do?

**Comment by jiriks74** [comment]
> Seems this is issue with DocumentServer
> 
> How did you installed DocumentServer on arm64? Currently we have no official support of this platform, did you use some third-party port?

No, I'm running naitive amd64 package. I changed dependencies to arm64 ones and I run the server itself with help of qemu

**Comment by ShockwaveNN** [comment]
I think until we support arm64 officially and natively anything can go wrong, especially with font generation scheme
So I don't think I can help you in any way

You can track our main issue for arm in [here](https://github.com/ONLYOFFICE/DocumentServer/issues/152)

**Comment by jiriks74** [comment]
@ShockwaveNN 
Where does the server take fonts from? I'd copy the font's I want there and tried it that way

**Comment by jiriks74** [comment]
Ok, I copied my fonts to `/usr/share/fonts/` to see if the script has problem getting the fonts through the subdirectories that is there

**Comment by ShockwaveNN** [comment]
> Where does the server take fonts from? I'd copy the font's I want there and tried it that way


You can take a look to `/bin/documentserver-generate-allfonts.sh` to see that font generation is doing

But AFAIR we got unresolved isssues for `allfontsgen` on arm

**Comment by jiriks74** [comment]
> > Where does the server take fonts from? I'd copy the font's I want there and tried it that way
> 
> You can take a look to `/bin/documentserver-generate-allfonts.sh` to see that font generation is doing
> 
> But AFAIR we got unresolved isssues for `allfontsgen` on arm

But that is making naitive arm package, right?

**Comment by ShockwaveNN** [comment]
Sorry, don't understand your question

Once again we do not support arm in any way so anything can be wrong

**Comment by jiriks74** [comment]
So far the fonts work, but I get error that system fonts will be used. Does that mean that it will use server's fonts or client's fonts?

**Comment by ShockwaveNN** [comment]
We do not use client's fonts in any way, so I think this mean server's default fonts will be used (around 30 of them)

**Comment by jiriks74** [comment]
> We do not use client's fonts in any way, so I think this mean server's default fonts will be used (around 30 of them)

Fixed. I needed windows fonts and I found [this](https://helpcenter.onlyoffice.com/installation/docs-community-install-fonts-linux.aspx)
>Helpful Hints

>ONLYOFFICE Docs always has a certain set of fonts with it. When installed it will check the presence of the following font files in the system: arial.ttf, calibri.ttf, cour.ttf, symbol.ttf, times.ttf, wingding.ttf.
>When a document is opened for the first time, ONLYOFFICE Docs will check which fonts are present in the document and uploads it from the computer with ONLYOFFICE Docs installation. In case the document contains fonts absent from the ONLYOFFICE Docs computer, it will upload the closest font substitute (the document layout and display might suffer from such substitution).
>In case you want to limit the number of fonts uploaded from the system, you can do the following:
> - delete fonts from the system and run the /usr/bin/documentserver-generate-allfonts.sh file, after that clear the browser cache and reopen the page (might ruin the original document layout and display);
> -  limit the number of fonts in documents used (using only most spread fonts: Times New Roman, Verdana, Tahoma, etc.);
> -  use compression (dynamic and static).

So I purged onlyoffice package, installed microsoft fonts, echoed my desired port, installed again and microsoft fonts are working
@ShockwaveNN

**Comment by ShockwaveNN** [comment]
Maybe your issue was not related to ARM at all, but was simple dpkg-configure issue, but I'm not sure

**Comment by jiriks74** [comment]
Ok, it does weird things now though. It shows Times New Roman in the dropdown menu but won't select it if I click on it and if I write it there and press enter it behaves like it does not exist.

**Comment by ShockwaveNN** [comment]
@jiriks74 Try to clear browser cache or open editor in Incognito mode

**Comment by jiriks74** [comment]
Oh, that works, thanks

**Comment by jiriks74** [comment]
@ShockwaveNN Last thing. I connected destkopeditors to nextcloud and if I click on something it opens browser instead of a new tab in desktopeditors. What could cause this?

**Comment by ShockwaveNN** [comment]
@jiriks74 Have no idea, once again on arm anything can be not working

**Comment by jiriks74** [comment]
> Have no idea, once again on arm anything can be not working
@ShockwaveNN 
Nope, this is client error I think. I logged out and back in and it looks like it works now

**Comment by fendaq** [comment]
@jiriks74 hi ,could you tell me how to run oo on arm64 debian? thx

**Comment by jiriks74** [comment]
> hi ,could you tell me how to run oo on arm64 debian? thx
@fendaq 
It's hacky a lot. And be prepared that presentation themes take a lot of time to generate (an hour almost for me).
Be careful not to break something on your system. Trying this a year ago, I almost broke the whole package system because I've created so many errors. Though this should work just fine.

Follow [this](https://helpcenter.onlyoffice.com/installation/docs-community-install-ubuntu.aspx) until the installation of onlyoffice docs. Then you have to follow this:
(I modified it from [here](https://github.com/ONLYOFFICE/DocumentServer/issues/152#issuecomment-689504125), because it didn't work for me)

1.    `sudo apt install qemu binfmt-support qemu-user-static -t buster-backports`
2.    If you want to have microsoft fonts working with the server `sudo apt install ttf-mscorefonts-installer`
3.    `sudo dpkg --add-architecture amd64`
4.    `sudo apt update && apt install libc6:amd64`
5.    `sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys CB2DE8E5`
6.    `echo "deb https://download.onlyoffice.com/repo/debian squeeze main" | sudo  tee /etc/apt/sources.list.d/onlyoffice.list`
7.    `sudo apt update && apt download onlyoffice-documentserver:amd64`
8.    `sudo dpkg-deb -R onlyoffice-documentserver*.deb output`
9.    `sudo nano output/DEBIAN/control`
10.    Modify the "Depends" line, making it look like this:
    `Depends: debconf:arm64 (>= 0.5) | debconf-2.0:arm64, adduser:arm64, ca-certificates:arm64, coreutils:arm64, curl:arm64, libasound2:arm64, libcairo2:arm64, libcurl3:arm64 | libcurl4:arm64, libcurl3-gnutls:arm64, libgconf-2-4:arm64, libgtk-3-0:arm64, libstdc++6:arm64 (>= 4.8.4), libxml2:arm64, libxss1:arm64, libxtst6:arm64, logrotate:arm64, mysql-client:arm64 | mariadb-client:arm64, nginx-extras:arm64 (>= 1.3.13), postgresql-client:arm64 (>= 9.1), pwgen:arm64, supervisor:all (>= 3.0b2), xvfb:arm64, zlib1g:arm64`
11.   ` sudo dpkg-deb -b output onlyoffice-documentserver-modified.deb`
12.    `sudo dpkg -i onlyoffice-documentserver-modified.deb`
13.    `sudo nano /var/lib/dpkg/info/onlyoffice-documentserver.postinst
14.    Put `exit 0` at the beginning of the file
15.    `sudo dpkg -i onlyoffice-documentserver-modified.deb`
16.    `sudo apt -f install`
17.    If this fails, uninstall onlyoffice `sudo apt purge onlyoffice-documentserver`, install the packages that are missing and repeat step 13
18.    `sudo nano /var/lib/dpkg/info/onlyoffice-documentserver.postinst
19.    Delete the exit 0 you've put there (may not be needed, I did it just in case)
20.    `sudo dpkg-reconfigure onlyoffice-documentserver `


Don't install updates. To update the documentserver `rm -rf onlyoffice-documentserver* output` and repeat from step 7
